<launch>
    <!-- unique camera name-->
    <arg name="camera_name" default="camera"/>
    <!-- Hardware depth registration -->
    <arg name="depth_registration" default="false"/>
    <arg name="serial_number" default=""/>
    <arg name="device_num" default="1"/>
    <arg name="vendor_id" default="0"/>
    <arg name="product_id" default="0"/>
    <arg name="enable_point_cloud" default="true"/>
    <arg name="enable_colored_point_cloud" default="false"/>
    <arg name="point_cloud_qos" default="default"/>
    <arg name="connection_delay" default="100"/>
    <arg name="color_width" default="640"/>
    <arg name="color_height" default="480"/>
    <arg name="color_fps" default="30"/>
    <arg name="enable_color" default="true"/>
    <arg name="flip_color" default="false"/>
    <arg name="color_qos" default="default"/>
    <arg name="color_camera_info_qos" default="default"/>
    <arg name="depth_width" default="640"/>
    <arg name="depth_height" default="480"/>
    <arg name="depth_fps" default="30"/>
    <arg name="enable_depth" default="true"/>
    <arg name="flip_depth" default="false"/>
    <arg name="depth_qos" default="default"/>
    <arg name="depth_camera_info_qos" default="default"/>
    <arg name="ir_width" default="640"/>
    <arg name="ir_height" default="480"/>
    <arg name="ir_fps" default="30"/>
    <arg name="enable_ir" default="true"/>
    <arg name="flip_ir" default="false"/>
    <arg name="ir_qos" default="default"/>
    <arg name="ir_camera_info_qos" default="default"/>
    <arg name="publish_tf" default="true"/>
    <arg name="tf_publish_rate" default="10.0"/>
    <arg name="ir_info_url" default=""/>
    <arg name="color_info_url" default=""/>
    <arg name="color_roi_x" default="-1"/>
    <arg name="color_roi_y" default="-1"/>
    <arg name="color_roi_width" default="-1"/>
    <arg name="color_roi_height" default="-1"/>
    <arg name="depth_roi_x" default="-1"/>
    <arg name="depth_roi_y" default="-1"/>
    <arg name="depth_roi_width" default="-1"/>
    <arg name="depth_roi_height" default="-1"/>
    <arg name="depth_scale" default="1"/>
    <arg name="color_depth_synchronization" default="false"/>
    <arg name="use_uvc_camera" default="true"/>
    <arg name="uvc_vendor_id" default="0x2bc5"/>
    <arg name="uvc_product_id" default="0x0501"/>
    <arg name="uvc_retry_count" default="100"/>
    <arg name="uvc_camera_format" default="mjpeg"/>
    <arg name="uvc_flip" default="false"/>
    <arg name="oni_log_level" default="verbose"/>
    <arg name="oni_log_to_console" default="false"/>
    <arg name="oni_log_to_file" default="false"/>
    <arg name="enable_d2c_viewer" default="false"/>
    <arg name="enable_publish_extrinsic" default="false"/>
    <group>
        <push-ros-namespace namespace="$(var camera_name)"/>
        <node name="camera" pkg="astra_camera" exec="astra_camera_node" output="screen">
            <param name="camera_name" value="$(var camera_name)"/>
            <param name="depth_registration" value="$(var depth_registration)"/>
            <param name="serial_number" value="$(var serial_number)"/>
            <param name="device_num" value="$(var device_num)"/>
            <param name="vendor_id" value="$(var vendor_id)"/>
            <param name="product_id" value="$(var product_id)"/>
            <param name="enable_point_cloud" value="$(var enable_point_cloud)"/>
            <param name="enable_colored_point_cloud" value="$(var enable_colored_point_cloud)"/>
            <param name="point_cloud_qos" value="$(var point_cloud_qos)"/>
            <param name="connection_delay" value="$(var connection_delay)"/>
            <param name="color_width" value="$(var color_width)"/>
            <param name="color_height" value="$(var color_height)"/>
            <param name="color_fps" value="$(var color_fps)"/>
            <param name="enable_color" value="$(var enable_color)"/>
            <param name="flip_color" value="$(var flip_color)"/>
            <param name="color_qos" value="$(var color_qos)"/>
            <param name="color_camera_info_qos" value="$(var color_camera_info_qos)"/>
            <param name="depth_width" value="$(var depth_width)"/>
            <param name="depth_height" value="$(var depth_height)"/>
            <param name="depth_fps" value="$(var depth_fps)"/>
            <param name="flip_depth" value="$(var flip_depth)"/>
            <param name="enable_depth" value="$(var enable_depth)"/>
            <param name="depth_qos" value="$(var depth_qos)"/>
            <param name="depth_camera_info_qos" value="$(var depth_camera_info_qos)"/>
            <param name="ir_width" value="$(var ir_width)"/>
            <param name="ir_height" value="$(var ir_height)"/>
            <param name="ir_fps" value="$(var ir_fps)"/>
            <param name="enable_ir" value="$(var enable_ir)"/>
            <param name="flip_ir" value="$(var flip_ir)"/>
            <param name="ir_qos" value="$(var ir_qos)"/>
            <param name="ir_camera_info_qos" value="$(var ir_camera_info_qos)"/>
            <param name="publish_tf" value="$(var publish_tf)"/>
            <param name="tf_publish_rate" value="$(var tf_publish_rate)"/>
            <param name="ir_info_url" value="$(var ir_info_url)"/>
            <param name="color_info_url" value="$(var color_info_url)"/>
            <param name="color_roi_x" value="$(var color_roi_x)"/>
            <param name="color_roi_y" value="$(var color_roi_y)"/>
            <param name="color_roi_width" value="$(var color_roi_width)"/>
            <param name="color_roi_height" value="$(var color_roi_height)"/>
            <param name="depth_roi_x" value="$(var depth_roi_x)"/>
            <param name="depth_roi_y" value="$(var depth_roi_y)"/>
            <param name="depth_roi_width" value="$(var depth_roi_width)"/>
            <param name="depth_roi_height" value="$(var depth_roi_height)"/>
            <param name="depth_scale" value="$(var depth_scale)"/>
            <param name="color_depth_synchronization" value="$(var color_depth_synchronization)"/>
            <param name="use_uvc_camera" value="$(var use_uvc_camera)"/>
            <param name="uvc_vendor_id" value="$(var uvc_vendor_id)"/>
            <param name="uvc_product_id" value="$(var uvc_product_id)"/>
            <param name="uvc_retry_count" value="$(var uvc_retry_count)"/>
            <param name="uvc_camera_format" value="$(var uvc_camera_format)"/>
            <param name="uvc_flip" value="$(var uvc_flip)"/>
            <param name="oni_log_level" value="$(var oni_log_level)"/>
            <param name="oni_log_to_console" value="$(var oni_log_to_console)"/>
            <param name="oni_log_to_file" value="$(var oni_log_to_file)"/>
            <param name="enable_d2c_viewer" value="$(var enable_d2c_viewer)"/>
            <param name="enable_publish_extrinsic" value="$(var enable_publish_extrinsic)"/>
            <remap from="/$(var camera_name)/depth/color/points" to="/$(var camera_name)/depth_registered/points"/>
        </node>
    </group>
</launch>
